[DCOS-40332] Integration tests for local profile volume support.

chhsia0 opened the pull request:
This PR adds an integration test for #2707. Changes include:
* Added a `hello.volume_profile` option to the Hello World service.
* Added an example YAML to test profile mount volumes.
* Created an additional MOUNT volume with profile "xfs" on every agent for the test.

chhsia0 commented:
@nickbp Removed the WIP tag. The test passed locally. PTAL!

nickbp commented on frameworks/helloworld/tests/test_profile_mount_volumes.py:
```diff
@@ -0,0 +1,32 @@
+@pytest.mark.sanity
+@pytest.mark.dcos_min_version("1.12")
+def test_profile_mount_volumes():
+    sdk_tasks.check_running(config.SERVICE_NAME, NUM_HELLO)
```
Would it be possible to verify here that the correct `xfs` volumes were indeed consumed? e.g. by fetching and looking at state.json or slaves.json. For some examples of accessing mesos jsons, see `sdk_tasks.py`, e.g.: `sdk_cmd.cluster_request("GET", "/mesos/tasks").json()["tasks"]`

chhsia0 replied:
The task itself already checked the filesystem (`df -t $VOLUME_PROFILE`). Wouldn't that be sufficient?

nickbp replied:
Oh I missed that. I think that's fine.

nickbp commented on tools/create_testing_volumes.py:
```diff
@@ -58,11 +61,41 @@ def mount_volumes():
         )
 
+    # To create profile mount volumes, we manually run `make_disk_resources.py`
+    # to generate disk resources, then use `jq` to parse the result and set the
```
Comment out of date: not using `jq` after all

chhsia0 commented:
Added a commit to address the comments and fix testing issues:
1. Use `/opt/mesosphere/bin/python` instead of `python` which may not exist on the cluster (The CI nodes have `python3` instead).
2. Updated the example YAML to use different container paths for task volumes. Otherwise the latter one will overwrite the first one and will make the test flaky.
3. Updated the example YAML to test the filesystem of the pod volume as well.
